genesis/model/resolver/child.go

```diff
@@ -47,7 +50,11 @@ func (r *childResolver) GetObject(ref *object.Reference, classID string) (object
 		return nil, fmt.Errorf("object is not Proxy")
 	}
 
-	if proxy.GetClassID() != classID {
+	classIDString, ok := classID.(string)
```

Comment:
it is bad style to use types in names. Rename to classID. previous classID can be named like cls